### PR TITLE
Adapt to upcoming changes in HA Core 2022.3

### PR DESCRIPTION
**Pull request by emontnemery**
The native Spotify integration has seen some changes which will be included in HA Core 2022.3

This PR adapts to those changes, while being backwards compatible

Relevant HA Core PRs:
- https://github.com/home-assistant/core/pull/66285
- https://github.com/home-assistant/core/pull/66314

### CODE REVIEW

**Comment by emontnemery** [comment]
Not sure if backwards compatibility is wanted, or if the minimum version `hacs.json` should be bumped instead?

**Comment by fcusson** [comment]
Sorry about the delay, I was fairly swamped in the last few weeks. I'll start to review the code during the week. I will check if we can make the project work without requiring a version bump. If not possible, we'll simply bump it.

**Comment by fcusson** [comment]
@hmmbob can you provide more details on what's breaking?

**Comment by hmmbob** [comment]
The exact things Erik mentions:

> The breaking changes are exactly what's in this PR:
> 
> * `SpotifyMediaPlayer._devices` is renamed to `SpotifyMediaPlayer.entity.data.devices.data`
> * `SpotifyMediaPlayer._spotify` is renamed to `SpotifyMediaPlayer.data.client`
> 
> The PR now checks for `_devices` / `_ spotify`, and falls back to `.data.xxx` If there's no need to be backwards compatible, we should bump the HA Core version in `hacs,json` to `2022.3.0` and just access the `.data.xxx`

**Comment by hmmbob** [comment]
The fix works, running it locally on 2022.3.0b1 and issue is solved.

**Comment by fcusson** [comment]
@hmmbob thank you for the details. I'm working on a potential way to have retrocompatibility while using the new method. If it's not possible, we will simply bump the home assistant version to 2022.3 like @emontnemery mentionned

**Comment by fcusson** [comment]
@emontnemery after testing, we have no breaking changes with your method. Code is working in 2022.2. Actually the if _devices logic you included isn't necessary even. For retrocompatibility, especially if we have a user with a very old version that doesn't have access to the new method you access, we'll keep error handling. I'll send a code review, there is a few changes I would like to see first.

File: custom_components/spotcast/helpers.py [hunk 1]
```diff
@@ -28,15 +28,24 @@ def get_spotify_devices(hass, spotify_user_id):
                 isinstance(entity, SpotifyMediaPlayer)
                 and entity.unique_id == spotify_user_id
             ):
-                _LOGGER.debug(
-                    f"get_spotify_devices: {entity.entity_id}: {entity.name}: %s",
-                    entity._devices,
-                )
+                if hasattr(entity, "_devices"):
```
Comment (fcusson) [inline custom_components/spotcast/helpers.py:31]: modify to a try/except. The error type is `AttributeError`. I was thinking of something like that:

```python
try:
    devices_data = entity._devices
except(AttributeError):
    devices_data = entity.data.devices.data

_LOGGER.debug(f"get_spotify_devices: {entity.entity_id}: {entity.name}: %s", devices_data)
```

File: custom_components/spotcast/helpers.py [hunk 2]
```diff
@@ -28,15 +28,24 @@ def get_spotify_devices(hass, spotify_user_id):
...
+                    _LOGGER.debug(
+                        f"get_spotify_devices: {entity.entity_id}: {entity.name}: %s",
+                        entity.data.devices.data,
+                    )
                 spotify_media_player = entity
                 break
     if spotify_media_player:
         # Need to come from media_player spotify's sp client due to token issues
-        resp = spotify_media_player._spotify.devices()
+        if hasattr(entity, "_spotify"):
```
Comment (fcusson) [inline custom_components/spotcast/helpers.py:45]: encapsulate in a try/except also

```Python
try:
    resp = spotify_media_player._spotify.devices()
except(AttributeError):
    resp = spotify_media_player.data.client.devices()
```

**Review by fcusson — CHANGES_REQUESTED** [review]

**Comment by fcusson** [comment]
Due to the near approaching date for the 2022.03, I'll merge the request and make the change directly.

**Review by fcusson — APPROVED** [review]

**Pull request merged**